src/dask_histogram/core.py

```diff
@@ -729,41 +733,61 @@ def _reduction(
     ph: PartitionedHistogram,
     split_every: int | None = None,
 ) -> AggHistogram:
-    from dask.bag.core import empty_safe_aggregate
+    from dask.layers import DataFrameTreeReduction
 
     if split_every is None:
         split_every = dask.config.get("histogram.aggregation.split_every", 8)
     if split_every is False:
         split_every = ph.npartitions
 
     token = tokenize(ph, sum, split_every)
-    name = f"hist-aggregate-{token}"
-    k = ph.npartitions
-    b = ph.name
-    d = 0
-    dsk = {}
-    while k > split_every:
-        c = f"{name}{d}"
-        for i, inds in enumerate(partition_all(split_every, range(k))):
-            dsk[(c, i)] = (
-                empty_safe_aggregate,
-                sum,
-                [(b, j) for j in inds],
-                False,
-            )
-        k = i + 1
-        b = c
-        d += 1
-    dsk[(name, 0)] = (
-        empty_safe_aggregate,
-        sum,
-        [(b, j) for j in range(k)],
-        True,
+
+    label = "histreduce"
+
+    name_comb = f"{label}-combine-{token}"
+    name_agg = f"{label}-agg-{token}"
+
+    dftr = DataFrameTreeReduction(
+        name=name_agg,
+        name_input=ph.name,
+        npartitions_input=ph.npartitions,
+        concat_func=sum,
+        tree_node_func=lambda x: x,
```

Comment:
I would NOT have guessed this, the naming of these functions is odd. My guess was concat_func=tuple, tree_node_func=sum

Reply:
wow, that's incredible...